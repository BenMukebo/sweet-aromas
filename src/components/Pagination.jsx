import React from 'react';

export default function Pagination() {
  const pages = [1, 2, 3, 4, 5];
  return (
    <div className="pagination-main-container">
      <ul>
        {pages.map((page) => <li key={page}>{page}</li>)}
      </ul>
    </div>
  );
}
