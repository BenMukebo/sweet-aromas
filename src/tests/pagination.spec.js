import renderer from 'react-test-renderer';

import Pagination from '../components/Pagination';

describe('Pagination', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Pagination
          itemsCount={30}
          pageSize={6}
          currentPage={1}
          onPageChange={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
