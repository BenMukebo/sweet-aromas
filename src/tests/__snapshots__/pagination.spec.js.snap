// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render correctly 1`] = `
<div
  className="pagination-main-container d-flex flex-center"
>
  <ul
    className="d-flex"
  >
    <button
      className="pagination-item active"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="pagination-item"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="pagination-item"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="pagination-item"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="pagination-item"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
  </ul>
</div>
`;
